src/jit/lowerarmarch.cpp

```diff
@@ -777,7 +777,15 @@ void Lowering::ContainCheckSIMD(GenTreeSIMD* simdNode)
             op2 = simdNode->gtOp.gtOp2;
 
             // If the index is a constant, mark it as contained.
-            CheckImmedAndMakeContained(simdNode, op2);
+            if (op2->IsCnsIntOrI())
+            {
+                MakeSrcContained(simdNode, op2);
+            }
+            else
+            {
+                // If the index is not a constant, we will use the SIMD temp location to store the vector.
+                comp->getSIMDInitTempVarNum();
```

Comment:
I notice that xarch calls `getSIMDInitTempVarNum` in TreeNodeInfoInitSIMD. Is there a reason why you call it here instead?

Reply:
I can move it.  At the time it felt like a better place.  At the moment I see no reason for my initial opinion.